pkg/mhooks/event_type.go

```diff
@@ -46,20 +211,61 @@ type TransferCreated struct {
 
 type TransferUpdated struct {
 	// ID of the facilitator account
-	AccountID string `json:"accountID,omitempty"`
+	AccountID string `json:"accountID"`
 	// ID of the transfer
-	TransferID string `json:"transferID,omitempty"`
+	TransferID string `json:"transferID"`
 	// Status of the transfer
-	Status      string               `json:"status,omitempty"`
-	Source      PaymentMethodPartial `json:"source,omitempty"`
-	Destination PaymentMethodPartial `json:"destination,omitempty"`
+	Status      TransferUpdatedStatus `json:"status"`
+	Source      PaymentMethodPartial  `json:"source"`
+	Destination PaymentMethodPartial  `json:"destination"`
 }
 
+type TransferUpdatedStatus string
+
+const (
+	TransferUpdatedStatus_Pending   TransferUpdatedStatus = "pending"
+	TransferUpdatedStatus_Completed TransferUpdatedStatus = "completed"
+	TransferUpdatedStatus_Failed    TransferUpdatedStatus = "failed"
+	TransferUpdatedStatus_Reversed  TransferUpdatedStatus = "reversed"
+	TransferUpdatedStatus_Queued    TransferUpdatedStatus = "queued"
+	TransferUpdatedStatus_Canceled  TransferUpdatedStatus = "canceled"
+
+	TransferUpdatedStatus_SourceCompleted  TransferUpdatedStatus = "source.completed"
+	TransferUpdatedStatus_SourceCorrected  TransferUpdatedStatus = "source.corrected"
+	TransferUpdatedStatus_SourceInitiated  TransferUpdatedStatus = "source.initiated"
+	TransferUpdatedStatus_SourceReturned   TransferUpdatedStatus = "source.returned"
+	TransferUpdatedStatus_SourceOriginated TransferUpdatedStatus = "source.originated"
+	TransferUpdatedStatus_SourceConfirmed  TransferUpdatedStatus = "source.confirmed"
+	TransferUpdatedStatus_SourceSettled    TransferUpdatedStatus = "source.settled"
+	TransferUpdatedStatus_SourceFailed     TransferUpdatedStatus = "source.failed"
+	TransferUpdatedStatus_SourceCanceled   TransferUpdatedStatus = "source.canceled"
+
+	TransferUpdatedStatus_DestinationCompleted              TransferUpdatedStatus = "destination.completed"
+	TransferUpdatedStatus_DestinationCorrected              TransferUpdatedStatus = "destination.corrected"
+	TransferUpdatedStatus_DestinationInitiated              TransferUpdatedStatus = "destination.initiated"
+	TransferUpdatedStatus_DestinationReturned               TransferUpdatedStatus = "destination.returned"
+	TransferUpdatedStatus_DestinationOriginated             TransferUpdatedStatus = "destination.originated"
+	TransferUpdatedStatus_DestinationConfirmed              TransferUpdatedStatus = "destination.confirmed"
+	TransferUpdatedStatus_DestinationFailed                 TransferUpdatedStatus = "destination.failed"
+	TransferUpdatedStatus_DestinationAcceptedWithoutPosting TransferUpdatedStatus = "destination.accepted-without-posting"
+)
```

Comment:
Unfortunate that we have to do this; but due to not-so-great API design to support this feature. More context and proposed solution discussed w/ Product [here](https://moovfinancial.slack.com/archives/C01F61YE094/p1693421881511949)